account_proof.go

```diff
@@ -19,51 +19,46 @@
 package flow
 
 import (
+	"encoding/hex"
+	"errors"
 	"fmt"
+	"strings"
 
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
-type canonicalAccountProofWithoutTag struct {
-	Address   []byte
-	Timestamp uint64
-}
+// AccountProofNonceMinLenBytes is the minimum length of account proof nonces in bytes.
+const AccountProofNonceMinLenBytes = 32
 
-type canonicalAccountProofWithTag struct {
-	DomainTag []byte
-	Address   []byte
-	Timestamp uint64
-}
+// ErrInvalidNonce is returned when the account proof nonce passed to a function is invalid.
+var ErrInvalidNonce = errors.New("invalid nonce")
 
-// NewAccountProofMessage creates a new account proof message for singing. The appDomainTag is optional and can be left
-// empty. Note that the resulting byte slice does not contain the user domain tag.
-func NewAccountProofMessage(address Address, timestamp int64, appDomainTag string) ([]byte, error) {
-	var (
-		encodedMessage []byte
-		err            error
-	)
+type canonicalAccountProof struct {
+	AppID   string
+	Address []byte
+	Nonce   []byte
+}
 
-	if appDomainTag != "" {
-		paddedTag, err := padDomainTag(appDomainTag)
-		if err != nil {
-			return nil, fmt.Errorf("error encoding domain tag: %w", err)
-		}
+// NewAccountProofMessage creates a new account proof message for singing. The appID is optional and can be left
+// empty. The encoded message returned does not include the user domain tag.
```

Comment:
```suggestion
// NewAccountProofMessage creates a new account proof message for signing.T
//
// The returned encoded message is not prepended with the domain separation tag.
```

App ID is actually required now